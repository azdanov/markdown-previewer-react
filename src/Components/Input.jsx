// @flow

import React, { Component } from 'react';

type Props = {
  handleInputChange: Function,
  input: string,
};

class Input extends Component<Props> {
  props: {
    handleInputChange: Function,
    input: string,
  };
  handleChange = (event: SyntheticKeyboardEvent<HTMLTextAreaElement>) => {
    this.props.handleInputChange(event.currentTarget.value);
  };
  render() {
    return (
      <textarea
        value={this.props.input}
        onChange={this.handleChange}
        name="input-window"
      />
    );
  }
}

export default Input;
