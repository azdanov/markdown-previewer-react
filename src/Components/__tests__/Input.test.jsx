// @flow

import React from 'react';

import { shallow } from 'enzyme';
import Input from '../Input';

test('Input renders correctly', () => {
  const component = shallow(<Input input={''} handleInputChange={() => {}} />);
  expect(component).toMatchSnapshot();
});

test('handleInputChange is called', () => {
  const handleInputChangeMock = jest.fn();
  const inputWord = 'foo';

  const component = shallow(
    <Input input={''} handleInputChange={handleInputChangeMock} />,
  );

  component
    .find('textarea')
    .simulate('change', { currentTarget: { value: inputWord } });

  expect(handleInputChangeMock).toBeCalled();
  expect(handleInputChangeMock).toBeCalledWith(inputWord);
});
