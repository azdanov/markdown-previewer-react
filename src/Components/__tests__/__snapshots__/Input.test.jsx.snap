// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<textarea
  className="input"
  name="input-window"
  onChange={[Function]}
  value=""
/>
`;
